proj/src/main/webapp/static/signup.html

```diff
@@ -55,18 +41,17 @@ <h4>welcome</h4>
           <div class="row justify-content-md-center">
             <form>
               <label for="fname">first name:</label><br>
-              <input type="text" id="fname" name="fname" class="input"><br>
+              <input type="text" id="fname" name="fname" class="input" required><br>
               <label for="lname">last name:</label><br>
-              <input type="text" id="lname" name="lname" class="input"><br>
+              <input type="text" id="lname" name="lname" class="input" required><br>
               <label for="email">email:</label><br>
               <input type="text" id="email" name="email" class="input" required><br>
               <label for="password">password:</label><br>
-              <input type="password" id="pass" name="password" class="input" required><br>
+              <input type="text" id="password" name="password" class="input" required><br>
```

Comment:
Shouldn't this be `type="password"`? https://www.w3schools.com/tags/att_input_type_password.asp

Reply:
It should be. I updated it in a recent pull request.